scripts/add-chain.sh

```diff
@@ -1,161 +1,15 @@
-  SystemConfigProxy=$(jq -j .SystemConfigProxy $DEPLOYMENTS_DIR/.deploy)
-else # use legacy deployment artifact schema
-  AddressManager=$(jq -j .address $DEPLOYMENTS_DIR/AddressManager.json)
-  L1CrossDomainMessengerProxy=$(jq -j .address $DEPLOYMENTS_DIR/L1CrossDomainMessengerProxy.json)
-  L1ERC721BridgeProxy=$(jq -j .address $DEPLOYMENTS_DIR/L1ERC721BridgeProxy.json)
-  L1StandardBridgeProxy=$(jq -j .address $DEPLOYMENTS_DIR/L1StandardBridgeProxy.json)
-  L2OutputOracleProxy=$(jq -j .address $DEPLOYMENTS_DIR/L2OutputOracleProxy.json)
-  OptimismMintableERC20FactoryProxy=$(jq -j .address $DEPLOYMENTS_DIR/OptimismMintableERC20FactoryProxy.json)
-  SystemConfigProxy=$(jq -j .address $DEPLOYMENTS_DIR/SystemConfigProxy.json)
-  OptimismPortalProxy=$(jq -j .address $DEPLOYMENTS_DIR/OptimismPortalProxy.json)
-  SystemConfigProxy=$(jq -j .address $DEPLOYMENTS_DIR/SystemConfigProxy.json)
-fi
-
-# scrape remaining address live from the chain
-SuperchainConfig=$(cast call $OptimismPortalProxy "superchainConfig()(address)" -r $L1_RPC_URL) || "" # first command could fail if bedrock
-Guardian=$(cast call $SuperchainConfig "guardian()(address)" -r $L1_RPC_URL) || $(cast call $OptimismPortalProxy "GUARDIAN()(address)" -r $L1_RPC_URL) # first command could fail if bedrock
-Challenger=$(cast call $L2OutputOracleProxy "challenger()(address)" -r $L1_RPC_URL) # first command could fail if FPAC: TODO figure out how to get this role if FPAC
-ProxyAdmin=$(jq -j .address $DEPLOYMENTS_DIR/ProxyAdmin.json)
-ProxyAdminOwner=$(cast call $ProxyAdmin "owner()(address)" -r $L1_RPC_URL)
-SystemConfigOwner=$(cast call $SystemConfigProxy "owner()(address)" -r $L1_RPC_URL)
-
-# add extra addresses data to registry
-mkdir -p $SUPERCHAIN_REPO/superchain/extra/addresses/$SUPERCHAIN_TARGET
-cat > $SUPERCHAIN_REPO/superchain/extra/addresses/$SUPERCHAIN_TARGET/$CHAIN_NAME.json << EOF
-{
-  "AddressManager": "$AddressManager",
-  "L1CrossDomainMessengerProxy": "$L1CrossDomainMessengerProxy",
-  "L1ERC721BridgeProxy": "$L1ERC721BridgeProxy",
-  "L1StandardBridgeProxy": "$L1StandardBridgeProxy",
-  "L2OutputOracleProxy": "$L2OutputOracleProxy",
-  "OptimismMintableERC20FactoryProxy": "$OptimismMintableERC20FactoryProxy",
-  "OptimismPortalProxy": "$OptimismPortalProxy",
-  "SystemConfigProxy": "$SystemConfigProxy",
-  "ProxyAdmin": "$ProxyAdmin",
-  "ProxyAdminOwner": "$ProxyAdminOwner",
-  "SystemConfigOwner": "$SystemConfigOwner",
-  "Guardian": "$Guardian",
-  "Challenger": "$Challenger"
-}
-EOF
-
-
-# create genesis-system-config data
-# (this is deprecated, users should load this from L1, when available via SystemConfig).
-mkdir -p $SUPERCHAIN_REPO/superchain/extra/genesis-system-configs/$SUPERCHAIN_TARGET
-jq -r .genesis.system_config $ROLLUP_CONFIG > $SUPERCHAIN_REPO/superchain/extra/genesis-system-configs/$SUPERCHAIN_TARGET/$CHAIN_NAME.json
-
-
 # create extra genesis data
 mkdir -p $SUPERCHAIN_REPO/superchain/extra/genesis/$SUPERCHAIN_TARGET
 cd $MONOREPO_DIR
```

Comment:
UX nit that wasn't changed in this PR so feel free to postpone—we should save off the current dir, then `cd $originalDir` back to that after the below `go run` command. If the `go run` command can fail, we should put that `cd $originalDir` in a trap so it always runs

Reply:
I think the directory is only changed in a subshell when this script is run with `sh`? If I'm right then we don't need to make any changes here.

Reply:
Ah you might be right—I recall running a bash script recently were it was changed, but might be misremembering that situation, and I did not test this :)